### PR TITLE
북마크 기능 분리 완료

### LINKED ISSUE #614: 북마크 분리
## 💡 구현 기능
기존의 북마크 엔티티를 톡픽_북마크, 게임_북마크로 분리합니다.

### PR DESCRIPTION
**Pull request by gywns0417**
## 💡 작업 내용
- [x] `Bookmark` -> `TalkPickBookmark`, `GameBookmark` 분리
- [x] `resourceId` 필드, `BookmarkType` 클래스 삭제
- [x] 밸런스게임 세트 <-> 북마크_게임 연관관계 매핑
- [x] 톡픽 <-> 북마크_톡픽 연관관계 매핑

### postman 테스트 완료

기존의 `Bookmark` 클래스를 `TalkPickBookmark`, `GameBookmark` 클래스로 분리했습니다.
그리고 북마크 분리로 인해 필요없어진, `Bookmark` 내에 존재했던 `resourceId` 필드와 `BookmarkType` 클래스를 삭제했습니다.

## 🚩 후속 작업 (선택)
1. 게임 북마크 카운트 수가 취소 후에는 다시 올라가지 않는 이슈 해결 (active, deActive 활용)

## ✅ 셀프 체크리스트
- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요?
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels, Projects를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 테스트는 잘 통과했나요?
- [x] 불필요한 코드는 제거했나요?

closes #614

### CODE REVIEW

**Comment by Hanjaemo** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @gywns0417 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:exclamation: Error|3|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|
|New coverage|:white_check_mark: OK|0|< 80|
|New duplicated lines density|:white_check_mark: OK|0|> 10|
|New violations|:exclamation: Error|2|> 0|

[View on SonarQube](http://3.39.213.168:9000/dashboard?id=PICKO-BE)
###### _updated: 10/10/2024, 12:51:56 (UTC+0)_

**Comment by Hanjaemo** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @gywns0417 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:exclamation: Error|3|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|
|New coverage|:exclamation: Error|0|< 80|
|New duplicated lines density|:white_check_mark: OK|0|> 10|
|New violations|:exclamation: Error|5|> 0|

[View on SonarQube](http://3.39.213.168:9000/dashboard?id=PICKO-BE)
###### _updated: 10/10/2024, 13:00:12 (UTC+0)_

File: src/test/java/balancetalk/bookmark/application/TalkPickBookmarkServiceTest.java [hunk 1]
```diff
@@ -39,7 +39,10 @@ class BookmarkTalkPickServiceTest {
     BookmarkGenerator bookmarkGenerator;
 
     @Mock
-    BookmarkRepository bookmarkRepository;
+    TalkPickBookmarkRepository talkPickBookmarkRepository;
+
+    @Mock
+    GameBookmarkRepository gameBookmarkRepository;
```
Comment (Hanjaemo) [inline src/test/java/balancetalk/bookmark/application/TalkPickBookmarkServiceTest.java:45]: 이 테스트 코드에서는 gameBookmarkRepository를 사용하지 않는 것 같은데, 왜 mock으로 추가하셨을까요??

File: src/main/java/balancetalk/bookmark/domain/TalkPickBookmark.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @NotNull
+    private Boolean active;
+
+    public boolean matches(TalkPick talkPick) {
+        return isEqualsTalkPickId(talkPick);
+    }
+
+    private boolean isEqualsTalkPickId(TalkPick talkPick) {
+        return this.talkPick.equals(talkPick);
+    }
```
[the remark this answers is not in the document]
Comment (gywns0417) [reply src/main/java/balancetalk/bookmark/domain/TalkPickBookmark.java:38]: 감사합니다 😊

File: src/test/java/balancetalk/bookmark/application/TalkPickBookmarkServiceTest.java (hunk 1 above)
Comment (gywns0417) [reply src/test/java/balancetalk/bookmark/application/TalkPickBookmarkServiceTest.java:45]: 이거 이전 PR의 잔재입니다... 수정할게요!

**Comment by Hanjaemo** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @gywns0417 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:exclamation: Error|3|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|
|New coverage|:exclamation: Error|0|< 80|
|New duplicated lines density|:white_check_mark: OK|0|> 10|
|New violations|:exclamation: Error|5|> 0|

[View on SonarQube](http://3.39.213.168:9000/dashboard?id=PICKO-BE)
###### _updated: 10/10/2024, 14:35:56 (UTC+0)_

**Comment by Hanjaemo** [comment]
### SonarQube Quality Gate Result
- **Result**: :exclamation: Error
- Triggered by @gywns0417 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:exclamation: Error|3|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|
|New coverage|:exclamation: Error|0|< 80|
|New duplicated lines density|:white_check_mark: OK|0|> 10|
|New violations|:exclamation: Error|5|> 0|

[View on SonarQube](http://3.39.213.168:9000/dashboard?id=PICKO-BE)
###### _updated: 10/10/2024, 15:31:20 (UTC+0)_

**Review by Hanjaemo — APPROVED** [review]

**Pull request merged**